src/analysis/Environments/Environments.ts

```diff
@@ -350,43 +336,30 @@ export const Environments: React.FC<EnvironmentsProps> = (props) => {
     if (!_.some({ appName: errorAppId }, newApps)) {
       setErrorAppId(undefined);
     }
-    if (!_.some({ appName: deleteAppId }, newApps)) {
-      setDeleteAppId(undefined);
+    if (deleteAppModal.isOpen && !_.some({ appName: deleteAppModal.args?.appName }, newApps)) {
+      deleteAppModal.close();
     }
   });
   const loadData = withErrorIgnoring(refreshData);
 
   const pauseComputeAndRefresh = Utils.withBusyState(setLoading, async (compute: DecoratedComputeResource) => {
-    const wrappedPauseCompute = withErrorReporting('Error pausing compute', () =>
-      Utils.cond(
-        [
-          isRuntime(compute) && isGoogleWorkspaceInfo(compute.workspace),
-          () =>
-            ajax()
-              .Runtimes.runtimeWrapper(compute as RuntimeWithWorkspace)
-              .stop(),
-        ],
-        [
-          isRuntime(compute),
-          () =>
-            ajax()
-              .Runtimes.runtimeWrapper(compute as RuntimeWithWorkspace)
-              .stop(),
-        ],
-        [
-          isApp(compute) && isGoogleWorkspaceInfo(compute.workspace),
-          // @ts-expect-error
-          () => ajax().Apps.app(compute.workspace.googleProject, compute.appName).pause(),
-        ],
-        [
-          Utils.DEFAULT,
-          () => {
-            console.error('Pause is not currently implemented for azure apps');
-            return Promise.resolve();
-          },
-        ]
-      )
-    );
+    const wrappedPauseCompute = withErrorReporting('Error pausing compute', async () => {
```

Comment:
```suggestion
    const wrappedPauseCompute = withErrorReporting('Error pausing compute', () => {
```

This function doesn't need to be async, it doesn't await anything.

Reply:
technically it is for two reasons:
1) withErrorReporting expects an async (Promise function)
2) each of the calls to stop or pause are themselves async.

Reply:
Ok, if the type for `withErrorReporting` requires it. But re: 2, a function only needs to be `async` if it's `await`ing a Promise. It does not need to be `async` to call other functions that return Promises.